### ISSUE TITLE
Need an identifier for ExprienceEvents to indicated they represent the same original event observation

### ISSUE THREAD

**Issue by cdegroot-adobe** [adobe/xdm#428]
## What are the schemas that are affected by the issue
context/experienceevent

## What are examples of products that are impacted by the issue
Analytics, Launch

Need an identifier for ExprienceEvents to indicated they represent the same original event observation. For example a purchase event would cause several events to be raised in different systems that each represent this purchase, but may contain unique information generated by the different systems, where we have the opportunity to add a unique event id right at the event observation it can be included. When these events enter an XDM processing system they would be represented as ExperienceEvents and it would be very useful to have a standard place where this unique identifier be expressed. This would then inform a merging processor that creates a full representation of the purchase from all systems.

Note this is different to the context/experienceevent.@id field which is a way of identifying an instance of an experienceevent for future restating, lets say when further data enrichment is available.

**Comment by lrosenthol** [comment]
I don't understand how these are different from `@id` - sounds like EXACTLY what it is supposed to be used for...

**Comment by harleensahni** [comment]
Audience Manager needs this as well. If a customer has Analytics and Audience Manager and a feature enabled for server side forwarding, data collection happens through Analytics and is proxied to Audience Manager so that the client doesn't have to fire two similar calls on the page. This results in a data collection event in Analytics and and a data collection event in Audience Manager. Each of these systems will record different information in their respective events and mint their own unique event ids. But it's all part of the same logical event transaction. Does that help at all @lrosenthol?

**Comment by lrosenthol** [comment]
Is this "logical even transaction" information generated by a service somewhere?  How would you otherwise know that they are connected?

**Comment by harleensahni** [comment]
@lrosenthol, I'd think for the Analytics -> Audience Manager example that the first system, Analytics would mint the transaction id and pass it along to Audience Manager.

**Comment by cdegroot-adobe** [comment]
Let me give an example.
1. An end user executes a purchase on a shopping cart. At this moment an ID(id1) is generate for the event.
1(a). The advertising tracking observes the purchase event which flows into the ad tracking systems.
1(b). The mobile analytics tracking observes the purchase event which flows into the analytics systems.

2. The Advertising  system processes the purchase event (id1).
2(a). First it assigns a new ID to track enrichment of data to the event over time (id2), because there are trillions of events they are not upserted, new versions of the event are added as immutable inserts using the same id2, but newer write times in the event logs.
2(b). The event is enriched with attribution information.
2(c). The event is enriched with billing information.
3. Ad event is finally ready, it represents id1, has 3 instances of id2, of which the last one written is the authoritative one.

4. Similarly a process runs on the Analytics observation and it creates id3 to track this immutable progression of enrichment of the event.
4(a). enriches with ip2geo and initial device info.
4(b). enriches with updated device info.

A data platform is then tasked with bring in omni omni channel data and providing a complete view of the purchase in real time as all these enrichments are occurring.
It ingests the Ad(display) and Analytics datasets. 
They contain up to 6 records for the one purchase (ut=record update time)
id1, id2, ut0...data
id1, id2, ut1...data
id1, id2, ut2...data
id1, id3, ut0...data
id1, id3, ut1...data
id1, id3, ut2...data

The @id in experienceevent is the id2/3 in this case. This Issue is asking for a canonical representation of id1, the original observed event.  Very simular to the actual asset, where the asset schema represents the record of the asset, not the asset.

**Comment by lrosenthol** [comment]
Thanks @cdegroot-adobe .   Does that mean we just need a single element in these schemas (say `xdm:observedEventID`) or is there more to this?

**Comment by cdegroot-adobe** [comment]
Yes, it is the same sort of case as xdm:assetID. xdm:observedEventID is a great name.
